phase2/docker-images/toolbox/ansible/aws.yml

```diff
@@ -0,0 +1,179 @@
+- hosts: auto
+  tasks:
+
+  - name: Fetch instance identity document
+    uri:
+      url: "http://169.254.169.254/latest/dynamic/instance-identity/document"
+      method: "GET"
+      return_content: "yes"
+    register: ec2_instance_identity_document
+
+  - name: Store instance identity document
+    ## it doesn't return the right headers, so we have to parse it
+    set_fact:
+      ec2_instance_identity: "{{ ec2_instance_identity_document.content | from_json }}"
+
+  - name: Fetch all metadata about this instance
+    delegate_to: localhost
+    ec2_remote_facts:
+      region: "{{ ec2_instance_identity.region }}"
+      filters:
+        "instance-id": "{{ ec2_instance_identity.instanceId }}"
+    register: ec2_describe_instance
+
+  - name: Store cluster membership facts for this instance
+    set_fact:
+      kubernetes_cluster: "{{ ec2_describe_instance.instances[0].tags.KubernetesCluster }}"
+      kubernetes_role: "{{ ec2_describe_instance.instances[0].tags.Name }}"
+      pki_registry_domain: "{{ ec2_instance_identity.accountId }}.dkr.ecr.{{ ec2_instance_identity.region }}.amazonaws.com"
+      aws_local_hostname: "{{ ec2_describe_instance.instances[0].private_dns_name }}"
+      toolbox_image: "{{ lookup('env','KUBERNETES_ANYWHERE_TOOLBOX_IMAGE') }}"
+
+  - name: Fetch metadata about all other cluster members
+    delegate_to: localhost
+    ec2_remote_facts:
+      region: "{{ ec2_instance_identity.region }}"
+      filters:
+        "tag:KubernetesCluster": "{{ kubernetes_cluster }}"
+    register: ec2_describe_cluster
+
+  - name: Set cluster member list
+    set_fact:
+      kubernetes_cluster_memebers: "{{ ec2_describe_cluster.instances | map(attribute='private_ip_address') | ipv4 | list }}"
+
+  - name: Install all systemd units
+    copy: src="{{ item }}" dest="/etc/systemd/system/" owner="root" group="root" mode="0644"
+    with_fileglob:
+      - "/etc/toolbox/systemd-units/common/*"
+      - "/etc/toolbox/systemd-units/ec2/*"
+      - "/etc/toolbox/systemd-units/with-pki/*"
+
+  - file: path=/etc/kubernetes-anywhere state=directory owner="root" group="root" mode="0755"
+
+  - name: Set hostname for kubelet to use
+    copy: dest="/etc/kubernetes-anywhere/local.env"
+                mode="0644" owner="root" group="root"
+                content="AWS_LOCAL_HOSTNAME=\"{{ aws_local_hostname }}\"\nWEAVE_KNOWN_PEERS=\"{{ kubernetes_cluster_memebers | join(' ') }}\"\n"
+
+  - name: Load new systemd units
+    shell: systemctl daemon-reload
+
+  - name: Enable base systemd units
+    shell: systemctl enable \
```

Comment:
I wasn't able to verify whether `service` module supports `.target` units and wasn't sure it'll call `service` command or `systemctl`, so I've just done this in a well-know-working option. I've looked at `systemd` module and seemed not quite mature enough at this point, but I'll take a look at `service` module source code now.